### ISSUE TITLE
HIMS Braille Emotion display sometimes failing to be detected when connected to USB

### ISSUE THREAD

**Issue by christiancomaschi** [nvaccess/nvda#18114]
### Steps to reproduce:
- connect a HIMS BrailleEmotion display to USB port if not already connected
- Start NVDA with HIMS Braille display selected
- NVDA should recognize it

### Actual behavior:
- NVDA sometimes does not recognize it, especially when the device is seen also as a storage drive

### Expected behavior:

NVDA should recognize it

### NVDA logs, crash dumps and other attachments:
Log attached
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
NVDA 2024.4.2
#### Windows version:
Windows 10, latest update

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA Portable 2025 Beta1, same behavior

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

[nvda_braille_display_not_detected.log](https://github.com/user-attachments/files/20226755/nvda_braille_display_not_detected.log)

**Comment by dreinn** [comment]
Same problem with a Focus By Freedom Scientific.
NVDA tries connecting to the display braille, but it select a wrong USB-ID, and all fail.
If necessary, I can put a log also of this.

**Comment by christiancomaschi** [comment]
> Same problem with a Focus By Freedom Scientific. NVDA tries connecting to the display braille, but it select a wrong USB-ID, and all fail. If necessary, I can put a log also of this.

Let's hope the fix I have submitted, if approved, makes also your Focus work properly!
